src/lib/utils/error-logger.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const logger = {};
+
+const getLogData = (err) => {
+    let logData;
+    if (err.stack) {
+        logData = err.stack;
+    } else if (err.message) {
+        logData = err.message;
+    } else if (err.toString instanceof Function) {
+        logData = err.toString();
+    } else {
+        logData = err;
+    }
+
+    return logData;
+};
+
+module.exports.register = (server, options, next) => {
+
+    //logs error
+    logger.logError = (tags,err) => {
+        server.log(['error'].concat(tags), getLogData(err));
+    };
+
+    //logs and throws error
+    logger.logAndThrowError = (tags,err) => {
```

Comment:
Respectfully, this seems somewhat out-of-order to me. the server should catch unhandled error events _regardless_ and log them.  therefore, I would expect routes to just `throw`, then if the error is _unhandled_, log, then crash.  creating a method to log and rethrow feels like anti-pattern to me.  I'm not declaring it so, but I don't see the value here.

in other words, if we 'just throw', we should still get logging.  consider, if we have some utility A that calls methods from utility B, and B errors!  Consider if _A handles B's errors_.  Should we error log?  I say _no, we shouldn't_, because A has handled it.  If we needed output, A's error handler should perhaps info log or something, if anything at all.  I'd say this strategy is preferred, because we shouldn't log in cases where the error is expected and handled.